### ISSUE TITLE
model overfiting

### ISSUE THREAD

**Issue by qrsforever** [confifu/RepNet-Pytorch#1]
Hello, when i train the model, after 35 epochs, the valid loss curve is as follows:

![Screenshot from 2021-06-03 10-10-50](https://user-images.githubusercontent.com/1296332/120576321-28c20400-c455-11eb-840f-b243002387d8.png)

Can you give me some tips ?

**Comment by confifu** [comment]
What datasets and loss functions did you use? There is also [UCFRep dataset](https://github.com/Xiaodomgdomg/Deep-Temporal-Repetition-Counting).

**Comment by qrsforever** [comment]
@confifu 

Used Countix Dataset:

> train dataset:  trainvids
> valid dataset:  valvids
> test dataset:  testvids

Loss:

SmoothL1Loss for `period length`

BCEWithLogitsLoss for `periodicity`

**Comment by confifu** [comment]
Given limited data, the model is bound to overfit. Use other datasets, the `SyntheticDataset` class implements the technique mentioned in the paper. There are some videos from kinetics dataset in the synthvid directory. I wrote the class so that it can take large single files as well, so you can put in any random long video files into it as well. Also compare the metrics (OBO and MAE) that you get with those of the paper. They are better indicator of model performance than absolute loss values.
